cmd/internal/cli/build_linux.go

```diff
@@ -220,16 +223,22 @@ func runBuild(cmd *cobra.Command, args []string) {
 
 func runBuildLocal(ctx context.Context, cmd *cobra.Command, dst, spec string, fakerootPath string) {
 	var keyInfo *cryptkey.KeyInfo
+	unprivilege := false
 	if buildArgs.encrypt || promptForPassphrase || cmd.Flags().Lookup("pem-path").Changed {
 		if namespaces.IsUnprivileged() {
-			sylog.Fatalf("You must be root to build an encrypted container")
+			unprivilege = true
 		}
 
 		k, err := getEncryptionMaterial(cmd)
 		if err != nil {
 			sylog.Fatalf("While handling encryption material: %v", err)
 		}
 		keyInfo = k
+
+		if keyInfo == nil && unprivilege {
+			sylog.Errorf("Please provide valid encryption info, please add `--pem-path` or `--passphrase`")
```

Comment:
I think it would be good to mention environment variables too, and not say "Please" twice.

I suggest
```
Missing encryption info, please add `--passphrase` or `--pem-path` or corresponding environment variable
```

Reply:
refected in this commit
https://github.com/apptainer/apptainer/pull/1185/commits/b2c0de740fb01d972ba614547cdf26c176d49cc8